clippy_lints/src/strings.rs

```diff
@@ -143,12 +153,12 @@ impl LintPass for StringLitAsBytes {
     }
 }
 
-impl<'a, 'tcx> LateLintPass<'a, 'tcx> for StringLitAsBytes {
-    fn check_expr(&mut self, cx: &LateContext<'a, 'tcx>, e: &'tcx Expr) {
-        use crate::syntax::ast::LitKind;
+impl EarlyLintPass for StringLitAsBytes {
+    fn check_expr(&mut self, cx: &EarlyContext<'_>, e: &crate::syntax::ast::Expr) {
```

Comment:
Is there a way to avoid having to use the full path here for Expr? It seemed like the Expr type for `LateLintPass` and `EarlyLintPass` are different, and this is the only way I could figure out to make them coexist.

Reply:
Add `use crate::syntax::ast;` to the top of the file and just write `ast::Expr` here. Also below you could use `ast::LitKind`	and `ast::ExprKind`

The Expr type of `LateLintPass` is `hir::Expr`, because these lints operate on the HIR. `EarlyLintPass` lints operate on the AST so the Expr type is `ast::Expr`.